airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/JobCreationAndStatusUpdateActivityImpl.java

```diff
@@ -261,6 +265,42 @@ public void reportJobStart(final ReportJobStartInput input) {
     }
   }
 
+  @Override
+  public void failNonTerminalJobs(final FailNonTerminalJobsInput input) {
+    log.info("failNonTerminalJobs called with input: {}", input);
+    try {
+      final Set<io.airbyte.scheduler.models.JobStatus> nonTerminalJobStatuses = Sets.difference(
+          Set.of(io.airbyte.scheduler.models.JobStatus.values()),
```

Comment:
Shouldn't we add a new static var here instead of calculatino the diff each an every time?

Reply:
I second this:  that set will never change at runtime and could be created once.

Reply:
Yep good point, created a static var for this in JobStatus